Simplenote/SimplenoteAppDelegate+Swift.swift

```diff
@@ -8,8 +8,52 @@ import CoreSpotlight
 extension SimplenoteAppDelegate {
 
     @objc
-    func makeCoreDataManager() throws -> CoreDataManager {
-        try CoreDataManager()
+    func setupStorage() {
+        let migrationResult = SharedStorageMigrator().performMigrationIfNeeded()
+
+        do {
+            try setupCoreData(migrationResult: migrationResult)
+        } catch {
+            fatalError(error.localizedDescription)
+        }
+    }
+
+    private func setupCoreData(migrationResult: MigrationResult) throws {
+        let settings = StorageSettings()
+
+        switch migrationResult {
+        case .notNeeded, .success:
+            try validateStorageDirectory(at: settings.sharedUserLibraryDirectory)
+            coreDataManager = try CoreDataManager(at: settings.sharedStorageURL)
+        case .failed:
+            try validateStorageDirectory(at: settings.legacyUserLibraryDirectory)
+            coreDataManager = try CoreDataManager(at: settings.legacyStorageURL)
+        }
+    }
```

Comment:
We've got a bit of duplication here, and we're not really doing anything different if the migration was `notNeeded / successful` or failed (all that changes is the URL!)

A possibility could be to:

1. Have `SharedStorageMigrator` return an instance of `StorageSettings` with the right URL
2. Update `StorageSettings` so that the URL to use is a calculated property in there